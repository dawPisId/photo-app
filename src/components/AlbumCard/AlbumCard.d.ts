
interface InputProps {
  name: string;
  albumNumber: number;
}

interface StateProps {
  albumId: number;
  id: number;
  title: string;
  url: string;
  thumbnailUrl: string;
}
