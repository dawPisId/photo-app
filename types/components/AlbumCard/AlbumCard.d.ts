/// <reference types="react" />
interface InputProps {
    name: string;
    albumNumber: number;
}
declare function AlbumCard({ name, albumNumber }: InputProps): JSX.Element;
export default AlbumCard;
