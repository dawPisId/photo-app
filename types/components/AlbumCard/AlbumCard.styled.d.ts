export declare const Layout: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const Label: import("styled-components").StyledComponent<"span", any, {}, never>;
export declare const ImageRow: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const Button: import("styled-components").StyledComponent<"div", any, {}, never>;
