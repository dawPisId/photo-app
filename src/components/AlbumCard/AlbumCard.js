import { Button, ImageRow, Label, Layout } from "./AlbumCard.styled";
import React, { useEffect, useState } from "react";

import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import { NavLink } from "react-router-dom";
import { URL } from "../../APIAddress";

function AlbumCard({ name, number }) {
  const [photos, setPhotos] = useState([]);
  useEffect(() => {
    fetch(`${URL}/albums/${number}/photos?_start=0&_end=3`)
      .then((response) => response.json())
      .then((responseData) => {
        setPhotos(responseData);
      })
      .catch((error) => {
        console.log("error", error);
      });
  }, [number]);
  return (
    <Layout>
      <Label>{name}</Label>
      <ImageRow>
        {photos.map((link, index) => (
          <img src={link.url} key={index} alt="" />
        ))}
      </ImageRow>
      <NavLink to={`/albums/${number}/photos/`}>
        <Button>
          SEE MORE
          <ArrowForwardIosIcon />
        </Button>
      </NavLink>
    </Layout>
  );
}
export default AlbumCard;
