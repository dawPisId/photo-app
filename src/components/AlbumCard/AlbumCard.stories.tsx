import { ComponentMeta, ComponentStory } from "@storybook/react";

import AlbumCard from "./AlbumCard";
import AlbumGrid from "../AlbumGrid/AlbumGrid";
import { BrowserRouter } from "react-router-dom";
import React from "react";

export default {
  title: "Photo App/AlbumCard",
  component: AlbumCard,
} as ComponentMeta<typeof AlbumCard>;

const Template: ComponentStory<typeof AlbumCard> = (args) => (
  <BrowserRouter>
    <AlbumGrid>
      <AlbumCard {...args} />
    </AlbumGrid>
  </BrowserRouter>
);

export const Main = Template.bind({});
Main.args = {
  name: "AlbumCard",
  albumNumber: 1,
};
