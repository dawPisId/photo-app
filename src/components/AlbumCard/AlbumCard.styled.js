import styled from "styled-components";

export const Layout = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  cursor: pointer;
  background-color: white;
  border-radius: 5px;
  box-shadow: 0px 5px 5px grey;
  padding: 0.5rem;
  height: auto;
  a {
    text-decoration: none;
    color: black;
    align-self: flex-end;
  }
`;

export const Label = styled.span`
  margin-top: 1rem;
  overflow: scroll;
  font-size: 1.5rem;
  font-family: "Roboto", sans-serif;
  &::-webkit-scrollbar {
    display: none;
  }
`;

export const ImageRow = styled.div`
  display: grid;
  grid-column-gap: 0.5rem;
  margin: 2rem 0rem 2rem 0rem;
  grid-template-columns: 1fr 1fr 1fr;
  align-items: center;
  img {
    width: 100%;
    height: 80%;
  }
`;

export const Button = styled.div`
  display: flex;
  border: 1px solid lightgrey;
  font-family: "Roboto", sans-serif;
  border-radius: 3px;
  font-size: 1rem;
  padding: 0.5rem;
  align-items: center;
  .MuiSvgIcon-root {
    margin-left: 0.5rem;
    font-size: 1rem;
  }
`;
