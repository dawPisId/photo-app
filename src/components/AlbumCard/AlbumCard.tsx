///<reference path="AlbumCard.d.ts"/>

import { Button, ImageRow, Label, Layout } from "./AlbumCard.styled";
import React, { useEffect, useState } from "react";

import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import { NavLink } from "react-router-dom";
import { URL } from "../../APIAddress";

const AlbumCard:React.FC<InputProps> = ({ name, albumNumber }) => {
  const [photos, setPhotos] = useState<StateProps[]>([]);
  useEffect(() => {
    fetch(`${URL}/albums/${albumNumber}/photos?_start=0&_end=3`)
      .then((response) => response.json())
      .then((responseData) => {
        setPhotos(responseData);
      })
      .catch((error) => {
        console.log("error", error);
      });
  }, [albumNumber]);
  return (
    <Layout>
      <Label>{name}</Label>
      <ImageRow>
        {photos.map((link, index) => (
          <img src={link.url} key={index} alt="" />
        ))}
      </ImageRow>
      <NavLink to={`/albums/${albumNumber}/photos/`}>
        <Button>
          SEE MORE
          <ArrowForwardIosIcon />
        </Button>
      </NavLink>
    </Layout>
  );
}
export default AlbumCard;
