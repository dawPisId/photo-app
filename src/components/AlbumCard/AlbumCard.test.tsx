import { render, waitFor } from "@testing-library/react";

import AlbumCard from "./AlbumCard";
import { BrowserRouter } from "react-router-dom";
import React from "react";

const mockResponseData = [
  {
    albumId: 1,
    id: 1,
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952",
    thumbnailUrl: "https://via.placeholder.com/150/92c952",
  },
  {
    albumId: 1,
    id: 2,
    title: "reprehenderit est deserunt velit ipsam",
    url: "https://via.placeholder.com/600/771796",
    thumbnailUrl: "https://via.placeholder.com/150/771796",
  },
  {
    albumId: 1,
    id: 3,
    title: "officia porro iure quia iusto qui ipsa ut modi",
    url: "https://via.placeholder.com/600/24f355",
    thumbnailUrl: "https://via.placeholder.com/150/24f355",
  },
  {
    albumId: 1,
    id: 4,
    title: "culpa odio esse rerum omnis laboriosam voluptate repudiandae",
    url: "https://via.placeholder.com/600/d32776",
    thumbnailUrl: "https://via.placeholder.com/150/d32776",
  },
];

it("renders correctly", async () => {
  const mockFetch = Promise.resolve({
    json: () => Promise.resolve(mockResponseData),
  });
  jest.spyOn(window, "fetch").mockImplementationOnce(() => mockFetch as any);
  const testRender = render(
    <BrowserRouter>
      <AlbumCard name="one" albumNumber={1} />
    </BrowserRouter>
  );
  await waitFor(() => testRender.getByTestId("imageRow"));
  expect(testRender).toMatchSnapshot();
});
