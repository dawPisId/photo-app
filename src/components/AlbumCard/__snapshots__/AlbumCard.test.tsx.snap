// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM juqWpf"
        data-testid="testAlbumCard"
      >
        <span
          class="sc-gtsrHT YXTaF"
        >
          one
        </span>
        <div
          class="sc-dlnjwi iYBmHw"
          data-testid="imageRow"
        >
          <img
            alt=""
            src="https://via.placeholder.com/600/92c952"
          />
          <img
            alt=""
            src="https://via.placeholder.com/600/771796"
          />
          <img
            alt=""
            src="https://via.placeholder.com/600/24f355"
          />
          <img
            alt=""
            src="https://via.placeholder.com/600/d32776"
          />
        </div>
        <a
          href="/albums/1/photos/"
        >
          <div
            class="sc-hKFxyN gSMifO"
          >
            SEE MORE
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM juqWpf"
      data-testid="testAlbumCard"
    >
      <span
        class="sc-gtsrHT YXTaF"
      >
        one
      </span>
      <div
        class="sc-dlnjwi iYBmHw"
        data-testid="imageRow"
      >
        <img
          alt=""
          src="https://via.placeholder.com/600/92c952"
        />
        <img
          alt=""
          src="https://via.placeholder.com/600/771796"
        />
        <img
          alt=""
          src="https://via.placeholder.com/600/24f355"
        />
        <img
          alt=""
          src="https://via.placeholder.com/600/d32776"
        />
      </div>
      <a
        href="/albums/1/photos/"
      >
        <div
          class="sc-hKFxyN gSMifO"
        >
          SEE MORE
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
